plugins/fields/text/tmpl/text.php

```diff
@@ -21,4 +23,13 @@
 	$value = implode(', ', $value);
 }
 
-echo htmlentities($value);
+$output = htmlentities($value);
+
+try
+{
+	echo HTMLHelper::_('InlineEditing.render', $item, $output, $field->name, 'com_fields');
```

Comment:
You should check if `$item` exists and has the desired format (array/stdClass).

Reply:
Yes, you should check before passing, just throwing a stone and see if the window breaks is not a good way :) Exceptions are exactly that exceptions, they should only be triggered in case of an unexpected situation. In this case you can check if you need to convert the type of $item.